corehq/apps/hqwebapp/views.py

```diff
@@ -211,6 +212,32 @@ def password_change(req):
     return render(req, template_name, {"form": password_form})
 
 
+# Corresponds to django.contrib.auth.views.password_reset_confirm()
+# Handles when a user clicks the link in their "reset password" e-mail and prompts for a new password
+def domain_reset_pwd(request, domain, uidb64, token, template_name='login_and_password/reset_pwd.html'):
+    domain_instance = Domain.get_by_name(domain)
+    if not domain_instance:
+        raise Http404
+    request.project = domain_instance
+
+    extra_context = {
+        'domain': domain,
+        'display_name': domain_instance.display_name() or domain,
+        'current_page': {'page_name': _('Welcome to %s!') % domain_instance.display_name()},
+    }
+    return password_reset_confirm(
+        request, uidb64, token, template_name,
+        set_password_form=HQSetPasswordForm, extra_context=extra_context
+    )
+
+
+def domain_reset_pwd_complete(request, domain,
+                              template_name='login_and_password/password_reset_complete.html',
+                              extra_context=None):
+    extra_context = {'domain_name': domain} if extra_context is None else dict(extra_context, domain_name=domain)
+    return password_reset_complete(request, template_name, extra_context)
```

Comment:
does this method validate that the password reset was actually completed? Otherwise, what happens if I go to this link without first going through the other steps?